### PR TITLE
Fetch results controller, adopt for the data model messages changes

### PR DESCRIPTION
**Pull request by mikeger**
### Issues

See https://github.com/wireapp/wire-ios-data-model/pull/534

## Requirements 

- [x] https://github.com/wireapp/wire-ios-data-model/pull/534
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/74
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/846

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationTableViewDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+    public var actionControllers: [String: ConversationMessageActionController] = [:]
+    
+    private func performFetch() {
+        createFetchController()
+        tableView.reloadData()
+    }
+    
+    private var readerPosition: Int = 0 {
+        didSet {
+            print("reader position is \(readerPosition)")
```
[the remark this answers is not in the document]
Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationTableViewDataSource.swift]: Yeah this is also not working yet :(

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m [hunk 2]
```diff
@@ -746,7 +726,8 @@ - (CGFloat)tableView:(UITableView *)tableView estimatedHeightForRowAtIndexPath:(
 
 - (NSIndexPath *)tableView:(UITableView *)tableView willSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    ZMMessage *message = [self.messageWindow.messages objectAtIndex:indexPath.section];
+
+    ZMMessage *message = (ZMMessage *)[self.dataSource.messages objectAtIndex:indexPath.row];
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m]: My understanding is that the messages indices correspond to `indexPath.section`, is that true? And if so, shouldn't we use the section index here?

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m]: You are right, this is a bug from the recent merge.

Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m]: I noticed some other places where this occurs, I'll mark them for you.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+PeekPop.swift [hunk 3]
```diff
@@ -31,10 +31,11 @@ extension ConversationContentViewController: UIViewControllerPreviewingDelegate
         let cellLocation = view.convert(location, to: tableView)
 
         guard let cellIndexPath = tableView.indexPathForRow(at: cellLocation),
-              let message = self.conversationMessageWindowTableViewAdapter.previewableMessage(at: cellIndexPath, in: tableView),
               let cell = tableView.cellForRow(at: cellIndexPath) as? SelectableView & UIView else {
             return .none
         }
+        
+        let message = self.dataSource.messages[cellIndexPath.row]
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+PeekPop.swift]: Also here?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+PinchZoom.m [hunk 4]
```diff
@@ -32,10 +32,10 @@ @implementation ConversationContentViewController (GestureRecognizerDelegate)
 - (nullable id<ZMConversationMessage>)messageAtPoint:(CGPoint)point
 {
     NSIndexPath *indexPath = [self.tableView indexPathForRowAtPoint:point];
-    if (indexPath == nil || indexPath.row >= (NSInteger)self.messageWindow.messages.count) {
+    if (indexPath == nil || indexPath.row >= (NSInteger)self.dataSource.messages.count) {
         return nil;
     }
-    id<ZMConversationMessage> message = [self.messageWindow.messages objectAtIndex:indexPath.row];
+    id<ZMConversationMessage> message = [self.dataSource.messages objectAtIndex:indexPath.row];
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+PinchZoom.m]: And here?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m [hunk 5]
```diff
@@ -555,7 +535,8 @@ - (void)updateVisibleMessagesWindow
     NSIndexPath *firstIndexPath = indexPathsForVisibleRows.firstObject;
     
     if (firstIndexPath) {
-        id<ZMConversationMessage>lastVisibleMessage = [self.messageWindow.messages objectAtIndex:firstIndexPath.section];
+        id<ZMConversationMessage>lastVisibleMessage = [self.dataSource.messages objectAtIndex:firstIndexPath.row];
+
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m]: Here too.

**Comment by mikeger** [comment]
@typfel this one is updated, I think we can merge it 👍 and cry later :hurtrealbad:

**Pull request merged**